ui/ExportQueryDialog.cpp

```diff
@@ -0,0 +1,81 @@
+#include "ExportQueryDialog.h"
+#include "ui_ExportQueryDialog.h"
+
+ExportQueryDialog::ExportQueryDialog(QStandardItemModel* model, QWidget *parent) :
+    QDialog(parent),
+    ui(new Ui::ExportQueryDialog),
+    m_model(model)
+{
+    ui->setupUi(this);
+    ui->textedit_preview->setFont(QFontDatabase::systemFont(QFontDatabase::FixedFont));
+    ui->buttonBox->button(QDialogButtonBox::Ok)->setDisabled(true);
+
+    connect(ui->linedit_outputFilePath, &QLineEdit::textChanged, this, &ExportQueryDialog::checkFilename);
+
+    connect(ui->combobox_delimiter, SIGNAL(currentIndexChanged(int)), this, SLOT(on_comboboxCurrentIndexChanged(int))); //weird, doesnt work with new signal/slot syntax
+    connect(ui->combobox_quote, SIGNAL(currentIndexChanged(int)), this, SLOT(on_comboboxCurrentIndexChanged(int)));
+    connect(ui->checkbox_includeHeader, &QCheckBox::stateChanged, this, &ExportQueryDialog::on_checkBoxStateChanged);
+
+    refreshText();
+}
+
+ExportQueryDialog::~ExportQueryDialog()
+{
+    delete ui;
+}
+
+void ExportQueryDialog::on_pushButton_released()
+{
+    m_outputFilePath = QFileDialog::getSaveFileName(this, "Save file", "~/", ".csv files (*.csv)");
```

Comment:
I'm not sure if "~/" works on windows. You could use QDir::homePath() as an alternative.